lib/wavefront-direct-reporter.js

```diff
@@ -0,0 +1,229 @@
+'use strict';
+var ScheduledReporter = require('metrics').ScheduledReporter,
+  Histogram = require('metrics').Histogram,
+  util = require('util'),
+  https = require('https'),
+  zlib = require('zlib'),
+  delta = require('./delta');
+
+const agent = new https.Agent({
+  keepAlive: true
+});
+
+/**
+ * A custom reporter that sends metrics to a wavefront server using the wavefrontf data format.
+ * @param {Report} registry report instance whose metrics to report on.
+ * @param {String} prefix A string to prefix on each metric (i.e. app.hostserver)
+ * @param {String} server The wavefront server of the form clusterName.wavefront.com.
+ * @param {String} token The Wavefront API token with direct ingestion permission.
+ * @param {object} tags The tags to add to wavefront measurements,defaults to none if not specified.
+ * @constructor
+ */
+function WavefrontDirectReporter(registry, prefix, server, token, globaltags, gzip) {
+  WavefrontDirectReporter.super_.call(this, registry);
+  this.prefix = prefix;
+  this.server = server;
+  this.token = token;
+  this.batchSize = 10000;
+  this.gzip = gzip;
+  this.globaltags = {};
+  if (!isEmpty(globaltags)) {
+    this.globaltags = globaltags;
+  }else{
+    this.globaltags = {};
+  }
+}
+
+util.inherits(WavefrontDirectReporter, ScheduledReporter);
+
+WavefrontDirectReporter.prototype.report = function() {
+  var metrics = this.getMetrics();
+  var self = this;
+  var points = [];
+
+  if(metrics.counters.length != 0) {
+    metrics.counters.forEach(function (count) {
```

Comment:
javascript has "arrow functions", which lexically bind `this`.
```javascript
metrics.counters.forEach(count => this.appendCounter(count, points));
```
See: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/Arrow_functions